docs/04-new-widgets.md

```diff
@@ -111,3 +111,10 @@ looks like this:
         widget:after_draw_child(context, cr, child_index, child, width, height)
     end
     widget:after_draw_children(context, cr, width, height)
+
+The `:get_children()` and `:set_children()` methods are called when setting a
+widget with the declarative layout system, therefore these two methods should be
+well defined. Specifically, `:get_children()` should return all the children to
+be able to draw them. `:set_children()` is called by the declarative system to
```

Comment:
I don't think anything in awesome currently ever calls `get_children` (and for drawing them, you also need a position and that's `:layout`'s job). @Elv13 ?